### ISSUE TITLE
Suddenly unable to work normally

### ISSUE THREAD

**Issue by chenjinxuan** [Kong/kong#5569]
kong version: 1.4.1
run env: kubernetes, kong-ingress-controller dbless

```
2020/02/14 00:15:49 [notice] 24#0: *892987 [lua] cache.lua:321: purge(): [DB cache] purging (local) cache, client: 127.0.0.1, server:kong_admin, request: "POST /config?check_hash=1 HTTP/2.0", host: "localhost:8444"
2020/02/14 00:15:49 [error] 24#0: *892987 [lua] events.lua:194: do_handlerlist(): worker-events: event callback failed; source=balancer, event=upstreams, pid=24 error='/usr/local/share/lua/5.1/kong/runloop/balancer.lua:451: attempt to index local 'upstream' (a nil value)
stack traceback:
--
2020/02/14 00:15:49 [error] 24#0: *892987 [lua] balancer.lua:635: do_target_event(): target create: upstream not found for e778b712-1f70-5ecc-86bb-3eb59f2e0510, client: 127.0.0.1, server: kong_admin, request: "POST /config?check_hash=1 HTTP/2.0", host: "localhost:8444"
2020/02/14 00:15:49 [error] 24#0: *892987 [lua] balancer.lua:641: do_target_event(): target create: balancer not found for ttxsapp.pt-services.80.svc, client: 127.0.0.1, server: kong_admin, request: "POST /config?check_hash=1 HTTP/2.0", host: "localhost:8444"
2020/02/14 00:15:49 [error] 24#0: *892987 [lua] balancer.lua:641: do_target_event(): target create: balancer not found for oms-api.pt-services.80.svc, client: 127.0.0.1, server: kong_admin, request: "POST /config?check_hash=1 HTTP/2.0", host: "localhost:8444"
2020/02/14 00:15:49 [error] 24#0: *892987 [lua] balancer.lua:641: do_target_event(): target create: balancer not found for cms-admin-api.cms.80.svc, client: 127.0.0.1, server: kong_admin, request: "POST /config?check_hash=1 HTTP/2.0", host: "localhost:8444"
127.0.0.1 - - [14/Feb/2020:00:15:49 +0000] "localhost:8444" "POST /config?check_hash=1 HTTP/2.0" 201 19140 "-" "Go-http-client/2.0" "-" "0.009" "/config" "POST" "-" - - - "-" "-" "-" "-" "-" -
```

**Comment by hbagdi** [comment]
Please upgrade to 1.4.3. 
1.4.2 has a couple of fixes which _might_ be related to this.

**Comment by hishamhm** [comment]
> Please upgrade to 1.4.3.

or even higher! 2.0.1 is out :)

**Comment by chenjinxuan** [comment]
thinks ,i get it

**Comment by chenjinxuan** [comment]
![image](https://user-images.githubusercontent.com/25172225/75470910-5d324f00-59cc-11ea-9064-d82e0cae48b7.png)
version: 1.5.0
It seems that this problem will still occur

**Comment by carnei-ro** [comment]
I'm in kong 2.0.3 and the error persists. Any updates?

**Comment by hbagdi** [comment]
@locao @javierguerragiraldez Anything to share here?

**Comment by dndx** [comment]
Hello,

Please give Kong 2.4 beta a try, which contains a lot of bug fixes of balancer since 2.0. You can find the package at https://bintray.com/kong.

If the problem persists feel free to reopen the issue. Closing it for now since it has been sitting for a while.

**Comment by work-jlsun** [comment]
@dndx    2.4 kong   still  have this problem

2021/05/19 08:25:29 [error] 36#0: *1110896 [lua] balancer.lua:624: get_balancer(): balancer not found for knative-sink-test-1601.default.80.svc, will create it, client: x.x.x.x server: kong, request: "POST /api/knative/test/sink HTTP/1.1", host: "knative-sink-test.default.example.com:31437"
2021/05/19 08:25:29 [error] 29#0: *1110873 [lua] balancer.lua:624: get_balancer(): balancer not found for knative-sink-test-1601.default.80.svc, will create it, client: x.x.x.x, server: kong, request: "POST /api/knative/test/sink HTTP/1.1", host: "knative-sink-test.default.example.com:31437"
2021/05/19 08:25:29 [error] 41#0: *1111238 [lua] balancer.lua:624: get_balancer(): balancer not found for knative-sink-test-1601.default.80.svc, will create it, client: x.x.x.x, server: kong, request: "POST /api/knative/test/sink HTTP/1.1", host: "knative-sink-test.default.example.com:31437"
